src/wp-includes/class-wp-admin-bar.php

```diff
@@ -12,28 +12,19 @@
  *
  * @since 3.1.0
  */
-#[AllowDynamicProperties]
```

Comment:
Since `__get()` is getting removed, it should be safe to remove this directive as well.

Reply:
No, let's leave this. Why? It's part of the larger scope of handling Dynamic Properties. The criteria for this attribute still exists with this class (see https://github.com/WordPress/wordpress-develop/commit/5ebe28966e5decbe1862c147bf98db0d8094038e which is [changeset 54133](https://core.trac.wordpress.org/changeset/54133))

>The logic used for this commit is as follows:
>* If a class already has the attribute: no action needed.
>* If a class does not extend: add the attribute.
>* If a class does extend:
>   * If it extends stdClass: no action needed (as stdClass supports dynamic properties).
>   * If it extends a PHP native class: add the attribute.
>   * If it extends a class from one of WP's external dependencies: add the attribute.
>*In all other cases: no action — the attribute should not be needed as child classes inherit from the parent.
>
>Whether or not a class contains magic methods has not been taken into account, as a review of the currently existing magic methods has shown that those are generally not sturdy enough and often even set dynamic properties (which they should not). See the [live stream from August 16, 2022](https://www.youtube.com/watch?v=vDZWepDQQVE) for more details.